README.md

```diff
@@ -399,235 +391,177 @@ function runQuery(query) {
     }
 }
 
-// promise style:
-const pool2 = new sql.ConnectionPool(config)
-const pool2Connect = pool2.connect()
-
-pool2.on('error', err => {
-    // ... error handler
-})
-
-function runStoredProcedure() {
-    return pool2Connect.then((pool) => {
-        pool.request() // or: new sql.Request(pool2)
-        .input('input_parameter', sql.Int, 10)
-        .output('output_parameter', sql.VarChar(50))
-        .execute('procedure_name', (err, result) => {
-            // ... error checks
-            console.dir(result)
-        })
-    }).catch(err => {
-        // ... error handler
-    })
+async function startApp() {
+    const config = {
+        user: '...',
+        password: '...',
+        server: 'localhost',
+        database: '...',
+        pool: {
+            max: 10,
+            min: 0,
+            idleTimeoutMillis: 30000
+        }
+    }
+    // customConnector is the only instance of this class used in the application
+    const customConnector = new CustomConnector(config);
+    await customConnector.init();
+    const queryResult = await customConnector.runQuery('SELECT TOP 10 * FROM table_name');
 }
 ```
 
-Awaiting or `.then`ing the pool creation is a safe way to ensure that the pool is always ready, without knowing where it
-is needed first. In practice, once the pool is created then there will be no delay for the next operation.
-
-As of v6.1.0 you can make repeat calls to `ConnectionPool.connect()` and `ConnectionPool.close()` without an error being
-thrown, allowing for the safe use of `mssql.connect().then(...)` throughout your code as well as making multiple calls to
-close when your application is shutting down.
-
-The ability to call `connect()` repeatedly is intended to make pool management easier, however it is still recommended
-to follow the example above where `connect()` is called once and using the original resolved connection promise.
-Repeatedly calling `connect()` when running queries risks running into problems when `close()` is called on the pool.
+To extend this example and make it more scaleable, you could include a check for failed connections and then call `sql.connect()` again to refresh the `globalPool` instance variable.
 
-**ES6 Tagged template literals**
+For Express applications, the following example uses a single global pool instance constant that is passed into the routes. The server start is then chained inside the `connect()` promise.
 
-```javascript
-new sql.ConnectionPool(config).connect().then(pool => {
-    return pool.query`select * from mytable where id = ${value}`
-}).then(result => {
-    console.dir(result)
-}).catch(err => {
-    // ... error checks
-})
+```js
+const express = require('express');
+const sql = require('mssql');
+const config  = {/*...*/};
+//instantiate a connection pool
+const globalPool = new sql.Connection(config);
+//require route handlers and use the same connection pool everywhere
+const route1 = require('./routes/route1')(globalPool);
```

Comment:
again, this is superfluous, the entire point of the `sql.connect()` helper is to provide easy access to this global pool, not for people to pass around objects as dependencies like this.

Whilst this is completely valid as an approach, I don't think it's one to be recommending to beginners who just want to drop-in the sql lib and get going.

I'm not express expert, but it looks like it may actually be better to use `app.locals` on the express app and place the pool there, then users don't need to worry about passing it around?